tests/test_user_stats_tracker.py

```diff
@@ -0,0 +1,35 @@
+import pytest
+
+from lecture_feedback.user_stats_tracker import UserStatsTracker, UserStatus
+
+
+def test_user_stats_tracker(monkeypatch: pytest.MonkeyPatch) -> None:
+    tracker = UserStatsTracker()
+
+    tracker.add_user("user1", UserStatus.GREEN)
+    tracker.add_user("user2", UserStatus.YELLOW)
+    tracker.add_user("user3", UserStatus.RED)
+
+    tracker.update_user_status("user1", UserStatus.RED)
+    tracker.set_user_active("user2")
+
+    stats = tracker.get_user_stats()
+    assert stats["user1"].status == UserStatus.RED
+    assert stats["user2"].status == UserStatus.YELLOW
+    assert stats["user3"].status == UserStatus.RED
+
+    red_count, yellow_count, green_count, unknown_count = tracker.get_status_counts()
+    assert red_count == 2
+    assert yellow_count == 1
+    assert green_count == 0
+    assert unknown_count == 0
+
+    import time
+
+    fake_time = [time.time()]
+    monkeypatch.setattr(time, "time", lambda: fake_time[0])
+
+    # advance time past the timeout
+    fake_time[0] += UserStatsTracker.USER_TIMEOUT_SECONDS + 1
+    tracker.clean_up_outdated_users()
+    assert len(tracker.get_user_stats()) == 0
```

Comment:
I think the test would be nicer if we add users ovr time (every second) and see that the first one drops out and the others are still in.
(currenlty clean_up_outdated_users could clear ALL users as soon as 1 is outdated)